fastcrypto-zkp/src/api.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::io::Cursor;
+
+use ark_bls12_381::Bls12_381;
+use ark_groth16::{Proof, VerifyingKey};
+use ark_relations::r1cs::SynthesisError;
+use ark_serialize::CanonicalDeserialize;
+
+use crate::conversions::BlsFr;
+use crate::verifier::{process_vk_special, verify_with_processed_vk};
+
+/// Verify Groth16 proof using the serialized representation of verifying key, input and proof.
+/// This assumes canoncical ser/de defined in Arkworks, instead of the Zcash bytes representation.
```

Comment:
@huitseeker which is the official serialization format, based on the current version of RFC?

Reply:
I support 1 with good documentation

Reply:
We can start with 1. and possibly ask for feedback from the community to then support 2, or, push comes to shove, expose the conversion functions themselves.